drivers/mfd/mfd_ad559x_i2c.c

```diff
@@ -35,7 +37,9 @@ static int mfd_ad559x_i2c_read_reg(const struct device *dev, uint8_t reg, uint8_
 
 	ARG_UNUSED(reg_data);
 
-	__ASSERT((reg & 0xf0) == 0, "reg bits [7:4] should be 0: 0x%x", reg);
+	if (reg >= AD559X_REG_SEQ_ADC || reg <= AD559X_REG_IO_TS_CONFIG) {
+		reg |= AD559X_REG_RD_POINTER;
+	}
```

Comment:
Just being curious, could you explain what this block does? I could not figure it out from the datasheet.

Reply:
I was actually looking at AD5592R and that was what confused me. Thanks for the explanation.